wandb/sync/sync.py

```diff
@@ -158,6 +165,16 @@ def _send_tensorboard(self, tb_root, tb_logdirs, send_manager):
         # send all of our records like a boss
         while not send_manager._interface.record_q.empty():
             data = send_manager._interface.record_q.get(block=True)
+            if len(data.history.ListFields()) != 0:
+                # this will always exist since tb_watcher injects global_step into each record
+                global_step = next(
+                    data.history.item[i].value_json
+                    for i in range(len(data.history.item))
+                    if data.history.item[i].key == "global_step"
+                )
```

Comment:
I know global_step can be null in raw tbrecords, I assume tb_watcher accounts for this and always adds a global_step?

Reply:
Good catch, it doesn't always, if the namespace is "" it does not add it